src/app/news.service.ts

```diff
@@ -10,16 +10,27 @@ export class NewsService {
   private _news$ = new Subject<string>();
   private _breakingNews$ = new Subject<string>();
 
-  news$ = this._news$.pipe(storeNews());
-  breakingNews$ = this._breakingNews$.pipe(storeNews());
+  news$ = this._news$.pipe(
+      storeNews(),
+      shareReplay()
+  );
 
-  allNews$ = combineLatest(
-    // combineLatest will start emitting when all of it's streams will emit at least one time
-    // that is why startWith() operator is required here - we want to emit normal news
-    // even when there were no breaking news yet and vice versa
-    this.breakingNews$.pipe(startWith([])),
-    this.news$.pipe(startWith([]))
-  ).pipe(map(([breaking, news]) => [...breaking, ...news]));
+  breakingNews$ = this._breakingNews$.pipe(
+      storeNews(),
+      shareReplay()
+  );
+
+  allNews$ = merge(
+    this.breakingNews$,
+    this.news$
+  ).pipe(
+      scan((allNews, news) => {
+          const lastNews = news[news.length - 1];
+
+          return [...allNews, lastNews];
+      }, []),
```

Comment:
I believe you can just merge `_news$` and `_breakingNews$` and then just apply `storeNews()` operator on it to get the same result.

But very nice anyway =)